### PR TITLE
Subclassed EmsCluster to ManageIQ::Providers::Microsoft::InfraManager::Cluster

### PR DESCRIPTION
**Pull request by chessbyte**

### CODE REVIEW

**Comment by agrare** [comment]
1. Need to update the persister collection name to `:clusters` [here](https://github.com/ManageIQ/manageiq-providers-scvmm/blob/master/app/models/manageiq/providers/microsoft/inventory/persister/infra_manager.rb#L4)
2. Need to update the parser to use the new `persister.clusters` [here](https://github.com/ManageIQ/manageiq-providers-scvmm/blob/master/app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb#L107), [here](https://github.com/ManageIQ/manageiq-providers-scvmm/blob/master/app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb#L270), and [here](https://github.com/ManageIQ/manageiq-providers-scvmm/blob/master/app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb#L300)
3. Update the refresher_spec assert_specific_cluster to check for the new `:type`

File: app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def parse_hosts
         :ems_ref          => data["ID"],
         :hostname         => data["Name"],
         :ipaddress        => identify_primary_ip(data),
-        :ems_cluster      => ems_cluster,
+        :cluster          => cluster,
```
Comment (agrare) [inline app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb]: This still needs to be `:ems_cluster`

File: app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb [hunk 2]
```diff
@@ -297,7 +297,7 @@ def parse_vms
         :location         => data["VMCPath"].blank? ? "unknown" : data["VMCPath"].sub(drive_letter, "").strip,
         :tools_status     => process_tools_status(data),
         :host             => persister.hosts.lazy_find(host_id),
-        :ems_cluster      => persister.ems_clusters.lazy_find(cluster_id),
+        :cluster          => persister.clusters.lazy_find(cluster_id),
```
Comment (agrare) [inline app/models/manageiq/providers/microsoft/inventory/parser/infra_manager.rb]: Same, only the `persister.clusters` had to change

File: spec/models/manageiq/providers/microsoft/infra_manager/refresher_spec.rb [hunk 3]
```diff
@@ -73,7 +73,7 @@ def assert_ems
     )
 
     expect(@ems.ems_folders.size).to eq(4) # HACK: Folder structure for UI a la VMware
-    expect(@ems.ems_clusters.size).to eq(1)
+    expect(@ems.clusters.size).to eq(1)
```
Comment (agrare) [inline spec/models/manageiq/providers/microsoft/infra_manager/refresher_spec.rb]: Should be `@ems.ems_clusters`

File: spec/models/manageiq/providers/microsoft/infra_manager/refresher_spec.rb [hunk 4]
```diff
@@ -165,7 +165,7 @@ def assert_specific_host
     )
 
     # @host2 = Host.find_by(:name => "SFBronagh.manageiq.com")
-    # expect(@host2.ems_cluster).to eq(@cluster)
+    # expect(@host2.cluster).to eq(@cluster)
```
Comment (agrare) [inline spec/models/manageiq/providers/microsoft/infra_manager/refresher_spec.rb]: Same

**Comment by agrare** [comment]
After those are resolved this repo is green for me with your other changes

**Comment by chessbyte** [comment]
@agrare I think I addressed all your comments.  Please re-review.

**Pull request merged**